src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import Tab from './Tab';
+
+const TabsDiv = styled.div`
+  text-transform: uppercase;
+  font-weight: bold;
+  user-select: none;
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  -ms-user-select: none;
+  ${({ vertical }) => {
+    if (vertical) {
+      return css`  
+    width: 25%;
+    display: table-cell;
+    margin-bottom: -1px;
+    text-align: left;`;
+    }
+    return css`  
+    display: flex;
+    flex-direction: row;
+    align-items: stretch;
+    flex: 1 1 auto;
+    text-align: center;
+    margin-bottom: -1px;
+    min-height: 53px;
+    text-align: center;
+    margin-right: -1.5px;`;
+  }}`
+  ;
+
+export default class Tabs extends React.Component {
+
+  static propTypes = {
+    /**
+     * Tab elements within container
+     */
+    children: PropTypes.node.isRequired,
+    /**
+     * Called when user selects the tab
+     */
+    onTabSelected: PropTypes.func,
+    /**
+     * The currently selected tab
+     */
+    selectedTabIndex: PropTypes.number,
+    /**
+     * Enable vertical tab layout
+     */
+    vertical: PropTypes.bool,
+  };
+
+  renderTabs = (children,
+                { onTabSelected, selectedTabIndex, vertical }) => React.Children.map(children, (child, index) => {
+      if (!React.isValidElement(child)) {
+        return child;
+      }
+
+      // rendering Tab
+      return (
```

Comment:
Based on the usage, it looks like you pass in Tab components as children, but then you wrap them in Tab again here. Won't that result in double-nested Tab structure?

I can see that you want this component to provide the `onClick` handler, `active`, `disabled`, `vertical` for child tabs. You could use `React.cloneElement` to provide those props to the Tab children that are already there.

Reply:
We do not have double-nested structure since we used child.props.children means we ignore the tag itself and replace it by Tab, which is not the best approach I believe

Switching to React.cloneElement